### PR TITLE
Add workflow to publish airflow packages to Test PyPI

### PR DESCRIPTION
**Pull request by gopidesupavan**
This will help us to publish the packages to Test PyPI repository along with dry runs. https://docs.pypi.org/trusted-publishers/using-a-publisher/

There are extra configs available https://github.com/marketplace/actions/pypi-publish#Non-goals we may add these as well go..and testing

### CODE REVIEW

File: .github/workflows/test-pypi-airflow-publish.yml [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        with:
+          name: ${{ inputs.artifact-name }}
+          merge-multiple: true
+          path: ./dist
+
+      - name: "Publishing ${{ needs.release-checks.outputs.publisher-name }} to Test PyPI"
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          packages-dir: "./dist"
+          repository-url: https://test.pypi.org/legacy/
```
Comment (gopidesupavan) [inline .github/workflows/test-pypi-airflow-publish.yml:156]: Test PyPI repo endpoint

**Review by potiuk — APPROVED** [review]

**Comment by gopidesupavan** [comment]
This is configured @potiuk would you like to press the button 😄  to publish to test PyPI?

**Comment by potiuk** [comment]
https://github.com/apache/airflow-publish/actions/runs/12421347590

**Comment by potiuk** [comment]
And published: https://test.pypi.org/project/apache-airflow/2.10.4/

**Comment by gopidesupavan** [comment]
Wooohooo... hurray it did attestation aswell 

https://search.sigstore.dev/?logIndex=156743856

**Comment by gopidesupavan** [comment]
https://test.pypi.org/project/apache-airflow/2.10.4/#apache_airflow-2.10.4.tar.gz

**Comment by potiuk** [comment]
NAAAAJS

**Comment by gopidesupavan** [comment]
Attestation default to true 😄 

https://github.com/pypa/gh-action-pypi-publish/blob/unstable/v1/action.yml#L83

**Comment by potiuk** [comment]
So,,,, seems we are ready to test it in 2.10.5 and maybe also for providers

**Comment by gopidesupavan** [comment]
> So,,,, seems we are ready to test it in 2.10.5 and maybe also for providers

Yes it looks we are green to test all providers :) 

To publish providers to Test PyPI, we need to register all projects with workflow. will get a test workflow for providers.

Could you please add me to providers projects in Test PyPI?

**Comment by potiuk** [comment]
> Could you please add me to providers projects in Test PyPI?

That is a bit problematic because of lack of organisations (but I see they added it when they added trusted publishing). I will do it shortly then and we can bring back test-pypi testing, because it seems that trusted publishing somewhat changed the approach of PyPI team and they brought back test-pypi to be first-class-citizen.

**Comment by potiuk** [comment]
Actually -> what we are going to have eventually we will have to move all our projects under "apache" organisation which is alraady created, after Trusted Publishing is done, so probably best will be to use test-pypi to test the new setup (`Apache Airflow` will have to become a team within the organisation). It will take a moment beceuse we will have to request the new organisation. I am on it.

**Comment by potiuk** [comment]
I asked the PyPI team to expedite organisation creation for "Test PyPI" (I have some contacts there), so as soon as this is done, I will arrange everything in the way that we should eventually expect in PyPI. Stay tuned @gopidesupavan :)

**Comment by gopidesupavan** [comment]
> Actually -> what we are going to have eventually we will have to move all our projects under "apache" organisation which is alraady created, after Trusted Publishing is done, so probably best will be to use test-pypi to test the new setup (`Apache Airflow` will have to become a team within the organisation). It will take a moment beceuse we will have to request the new organisation. I am on it.

Oh i see that makes it easier.

**Pull request merged**